background.js

```diff
@@ -68,134 +71,168 @@ function clean_amazon(url) {
+
+            // SEO friendly descriptiion + detail product
+            "*://*.amazon.com/*/dp/*",
+            "*://*.amazon.ca/*/dp/*",
+            "*://*.amazon.co.jp/*/dp/*",
+            "*://*.amazon.co.uk/*/dp/*",
+            "*://*.amazon.cn/*/dp/*",
+            "*://*.amazon.de/*/dp/*",
+            "*://*.amazon.fr/*/dp/*",
+            "*://*.amazon.in/*/dp/*",
+            "*://*.amazon.it/*/dp/*",
+            "*://*.amazon.com.mx/*/dp/*",
+            "*://*.amazon.com.au/*/dp/*",
+            "*://*.amazon.com.br/*/dp/*",
+
+            "*://*.amazon.com/gp/product/*",
+            "*://*.amazon.ca/gp/product/*",
+            "*://*.amazon.co.jp/gp/product/*",
+            "*://*.amazon.co.uk/gp/product/*",
+            "*://*.amazon.cn/gp/product/*",
+            "*://*.amazon.de/gp/product/*",
+            "*://*.amazon.fr/gp/product/*",
+            "*://*.amazon.in/gp/product/*",
+            "*://*.amazon.it/gp/product/*",
+            "*://*.amazon.com.mx/gp/product/*",
+            "*://*.amazon.com.au/gp/product/*",
+            "*://*.amazon.com.br/gp/product/*",
+        ],
+        types: ["main_frame"]
+    }, ["blocking"]
 );
 
 
-function remove_searchparams(requestDetails) {
-    var url = new URL(requestDetails.url);
-    console.debug('Entering remove_searchparams -- got url: ' + url);
-    if (url.search.length > 0) {
-        url.search = "";
-        // console.debug("Clean url to:", url.href);
-        return {redirectUrl: url.href};
-    }
-    return {redirectUrl: ''};
-}
+var remove_alisearchparams = build_query_param_remover(function (p) { return true });
 
 browser.webRequest.onBeforeRequest.addListener(
-    remove_searchparams,
+    remove_alisearchparams,
     {
         urls: [
             "*://*.aliexpress.com/item/*.html*",
             "*://*.aliexpress.com/store/product/*.html*",
+        ],
+        types: ["main_frame"]
+    }, ["blocking"]
+);
+
+var remove_fbcontentparam = build_query_param_remover(function (p) { return p == "efg" });
+browser.webRequest.onBeforeRequest.addListener(
+    remove_fbcontentparam,
+    {
+        urls: [
+            "*://*.fbcdn.net/*",
         ], types: ["main_frame"]
     }, ["blocking"]
 );
 
-function build_redirect_to_query_param(query_param_name){
-    // console.debug('Build redirect with params: ' + query_param_name);
-    const redirect_to_get_param = function(requestDetails){
+function build_redirect_to_query_param(query_param_name) {
+    const redirect_to_get_param = function (requestDetails) {
         const search_params = new URLSearchParams(new URL(requestDetails.url).search);
         const real_url_from_param = search_params.get(query_param_name);
-        if (real_url_from_param){
-            // console.debug('Redirecting to ' + real_url_from_param);
-            return {redirectUrl: real_url_from_param};
+        if (real_url_from_param) {
+            //console.log('Redirecting to ' + real_url_from_param);
```

Comment:
I generally like to use `debug()` for debugging messages instead of `log()`. I'm aware that there is not a big difference (so my jsdev friends say).